src/test/java/io/neonbee/test/endpoint/odata/ODataBatchTest.java

```diff
@@ -0,0 +1,168 @@
+package io.neonbee.test.endpoint.odata;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import java.io.ByteArrayInputStream;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import org.apache.olingo.server.api.deserializer.batch.BatchOptions;
+import org.apache.olingo.server.api.deserializer.batch.BatchRequestPart;
+import org.apache.olingo.server.core.deserializer.batch.BatchParser;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import io.neonbee.test.base.ODataEndpointTestBase;
+import io.neonbee.test.base.ODataRequest;
+import io.neonbee.test.endpoint.odata.verticle.TestService1EntityVerticle;
+import io.neonbee.test.helper.MultipartResponse;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.junit5.VertxTestContext;
+
+class ODataBatchTest extends ODataEndpointTestBase {
+
+    static Stream<Arguments> withValidKeys() {
+        return Stream.of(Arguments.arguments("id-1"));
+    }
+
+    static Stream<Arguments> withInvalidKeys() {
+        return Stream.of(Arguments.arguments("1337"));
+    }
+
+    @Override
+    protected List<Path> provideEntityModels() {
+        return List.of(TestService1EntityVerticle.getDeclaredEntityModel());
+    }
+
+    @BeforeEach
+    void setUp(VertxTestContext testContext) {
+        deployVerticle(new TestService1EntityVerticle())
+                .onComplete(testContext.succeedingThenComplete());
+    }
+
+    @Test
+    @DisplayName("Test serialization of ODataRequest to multipart batch body request")
+    void testBatchRequestSerialization(VertxTestContext testContext) {
+        testContext.verify(() -> {
+            String boundary = UUID.randomUUID().toString();
+            ODataRequest singleRequest = new ODataRequest(TestService1EntityVerticle.TEST_ENTITY_SET_FQN);
+            ODataRequest batchRequest = new ODataRequest(TestService1EntityVerticle.TEST_ENTITY_SET_FQN)
+                    .setBatch(boundary, singleRequest, singleRequest);
+
+            BatchParser batchParser = new BatchParser();
+            BatchOptions batchOptions = BatchOptions.with().isStrict(true).build();
+            ByteArrayInputStream inputStream = new ByteArrayInputStream(batchRequest.getBody().getBytes());
+            List<BatchRequestPart> partList =
+                    batchParser.parseBatchRequest(inputStream, boundary, batchOptions);
+            assertThat(partList).hasSize(2);
+
+            testContext.completeNow();
+        });
+    }
```

Comment:
What exactly is the purpose of this test? Is the purpose to check if Olingo can deserialize the request?

You have already verified the body serialization. This test would test the Olingo library itself, which is not necessary. That Olingo can deserialize the requests in general will be tested indirectly by the tests below.

Reply:
As discussed